### PR TITLE
Add first updater e2e test

### PR DESCRIPTION
**Pull request by raphaelgavache**
### What does this PR do?

Add a first simple test that verifies files ownership with the updater.
Tests are run manually with 
```
 E2E_PIPELINE_ID=GITLAB_ID  aws-vault exec sso-agent-sandbox-account-admin -- invoke new-e2e-tests.run --targets ./tests/updater
```
or on every change in /cmd/updater, /pkg/updaters, /tests/updater
[link to new-e2e-updater test run](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/438353475)

### Motivation

First of many tests, all updater behaviours will be verified in tests/updater in the following weeks

### CODE REVIEW

**Review by coignetp — APPROVED** [review]

File: test/new-e2e/pkg/environments/aws/host/host.go [hunk 1]
```diff
@@ -206,9 +216,13 @@ func Run(ctx *pulumi.Context, env *environments.Host, params *ProvisionerParams)
...
 		env.FakeIntake = nil
 	}
+	if !params.installUpdater {
+		// Suite inits all fields by default, so we need to explicitly set it to nil
+		env.Updater = nil
+	}
 
 	// Create Agent if required
-	if params.agentOptions != nil {
+	if params.agentOptions != nil && !params.installUpdater {
```
Comment (pducolin) [inline test/new-e2e/pkg/environments/aws/host/host.go:225]: **💬 suggestion**
Should you change the logic to
```go
if params.installUpdater  {
		// install updater
		// you might return error if params.agentOptions == nil
} else if params.agentOptions != nil {
  // install agent as usual
}
```
?

File: test/new-e2e/tests/updater/linux_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+func TestUpdaterSuite(t *testing.T) {
+	e2e.Run(t, &vmUpdaterSuite{}, e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake(
+		awshost.WithUpdater(),
+		awshost.WithEC2InstanceOptions(ec2.WithOSArch(os.UbuntuDefault, os.ARM64Arch)),
+	)))
+}
+
+func (v *vmUpdaterSuite) TestUpdaterOwnership() {
+	// updater user exists and is a system user
+	require.Equal(v.T(), "/usr/sbin/nologin\n", v.Env().RemoteHost.MustExecute(`getent passwd dd-updater | cut -d: -f7`))
```
Comment (pducolin) [inline test/new-e2e/tests/updater/linux_test.go:33]: **💬 suggestion**
You might add a log message to help investigations on error, for example
```suggestion
	require.Equal(v.T(), "/usr/sbin/nologin\n", v.Env().RemoteHost.MustExecute(`getent passwd dd-updater | cut -d: -f7`), "unexpected: user does not exist or is not a system user")
```

**Review by pducolin — COMMENTED** [review]
Only blocker is the `..gitlan-ci.yml` rule definition change, to run on main and release pipelines

File: test/new-e2e/tests/updater/linux_test.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ func TestUpdaterSuite(t *testing.T) {
 
 func (v *vmUpdaterSuite) TestUpdaterOwnership() {
 	// updater user exists and is a system user
-	require.Equal(v.T(), "/usr/sbin/nologin\n", v.Env().RemoteHost.MustExecute(`getent passwd dd-updater | cut -d: -f7`))
+	require.Equal(v.T(), "/usr/sbin/nologin\n", v.Env().RemoteHost.MustExecute(`getent passwd dd-updater | cut -d: -f7`), "unexpected: user does not exist or is not a system user")
 	// owner
 	require.Equal(v.T(), "dd-updater\n", v.Env().RemoteHost.MustExecute(`stat -c "%U" /opt/datadog/updater`))
```
Comment (pducolin) [inline test/new-e2e/tests/updater/linux_test.go:35]: **💬 suggestion**
You might add logs to other assertion too

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/pkg/environments/aws/host/host.go [hunk 4]
```diff
@@ -222,32 +222,33 @@ func Run(ctx *pulumi.Context, env *environments.Host, params *ProvisionerParams)
 	}
 
 	// Create Agent if required
-	if params.agentOptions != nil && !params.installUpdater {
-		agent, err := agent.NewHostAgent(awsEnv.CommonEnvironment, host, params.agentOptions...)
+	if params.installUpdater {
+		updater, err := updater.NewHostUpdater(awsEnv.CommonEnvironment, host, params.agentOptions...)
```
[the remark this answers is not in the document]
Comment (raphaelgavache) [reply test/new-e2e/pkg/environments/aws/host/host.go:226]: Yes it still installs without, I'm not currently using options

[the remark this answers is not in the document]
Comment (raphaelgavache) [reply test/new-e2e/pkg/environments/aws/host/host.go:226]: I'll add it though as when it's set to nil, it's expected that no agent is installed

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**